Address HDR UK comments - issue 206 (#242)
sa-tre/satre-specification: docs/source/pillars/data_management.md

  - You should have a system to help classify outputs.
  - Removing data from a TRE can be a difficult process as there is potential for sensitive data to be revealed.
    Having guidance, processes and methods will help ensure that outputs are correctly classified and, furthermore, that outputs due to be openly published are identified.

>     Encouraging openly published outputs will enhance a TRE’s impact and transparency.

Reviewer: ```suggestion
    Encouraging openly published outputs will enhance a TRE's impact and transparency.
```

  - Guidance
  - Importance
* - 3.4.1.

>   - You should provide an open metadata catalogue of available datasets for users.

Reviewer 2: ```suggestion
  - You should provide a metadata catalogue of available datasets for users.
```
`open` makes sense for TREs handling public data, but not necessarily for other data.

  - This is particularly relevant for TREs with population-level data collection of general interest.

>     This may not be appropriate for TREs where each project has its own data sharing agreement with one or more {ref}`information asset owner <data_roles>` or very sensitive datasets.

Reviewer 2: The `information asset owner` can delegate decisions on who can access it to another actor, so something like `data provider` makes sense. On the other hand this is guidance and is an example rather than a definitive statement, so maybe it doesn't matter?
